### PR TITLE
General tail stab

### PR DESCRIPTION
**Pull request by morrowwolf**
## About The Pull Request

Re-opening: https://github.com/cmss13-devs/cmss13/pull/954

With permission from Geeves I will be re-opening and maintaining some closed pull requests while he is on break.

## Why It's Good For The Game

"It's a cool-ish feature that sorta-ish used to exist, so this brings it back in a new-ish revamped-ish way. It's also something I loved using an AVP2010 when I kill my friends, so, nice to have."

## Changelog

:cl: Geeves
add: All xenos have a new generic ability, the Tail Stab, with a two tile range dealing 1.2 times the caste's maximum melee damage, making it more useful on the bigger xenos. It takes one second of being the selected ability before it can be used.
balance: Boilers have an advanced version of the tail stab, called the Toxic Tail Stab, where it injects the target with diluted molecular acid, causing burns and toxin damage.
/:cl:

### CODE REVIEW

**Comment by Chainsawicus** [comment]
Copying my latest feedback on the original PR - 

> Defender should receive a Tail Stab equivalent, it not receiving one just because the sprite has its tail as a mace-like appendage rather than a spear-like is an extremely weak argument for its exclusion. All you have to do is change the name on Defender specifically to "Tail Strike" or "Tail Crush" and leave the rest of the ability's effects as-is and it's a perfectly justifiable, fitting ability.
>
>I've seen people say that "tail sweep" is their tail stab, but it's not an equivalent whatsoever. Tail Sweep is a point-blank small non-damaging stun, while Tail Stab is a short range damaging attack that dazes the opponent and hits your targeted limb, it's not unreasonable to have two abilities that use the tail.

Besides that, thank you for re-opening this.

**Comment by Wintermote** [comment]
Didn't get to see it playtested, but I'd been hearing that this was giving xenos essentially a push-button free bone break on any given marine, is that intended?

File: code/modules/mob/living/carbon/xenomorph/abilities/boiler/boiler_powers.dm [hunk 1]
```diff
@@ -317,3 +317,9 @@
 	scatter = SCATTER_AMOUNT_TIER_1
 	bonus_projectiles_amount = 0
 	max_range = 4
+
+/datum/action/xeno_action/activable/tail_stab/boiler/use_ability(atom/A)
+	. = ..()
+	if(iscarbon(.))
+		var/mob/living/carbon/target = .
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/xenomorph/abilities/boiler/boiler_powers.dm:324]: bad for readability

File: code/modules/mob/living/carbon/xenomorph/abilities/general_powers.dm [hunk 2]
```diff
@@ -835,3 +835,88 @@
...
+	target.Daze(3)
+	shake_camera(target, 2, 1)
+
+	target.handle_blood_splatter(get_dir(owner.loc, target.loc))
+
+	apply_cooldown()
+	xeno_attack_delay(xeno)
+
+	..()
+	return target
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/xenomorph/abilities/general_powers.dm:922]: above

File: code/modules/power/lighting.dm [hunk 3]
```diff
@@ -593,6 +591,14 @@
 		sleep(1)
 		qdel(src)
 
+/obj/structure/machinery/light/handle_tail_stab(var/mob/living/carbon/Xenomorph/xeno)
+	if(is_broken())
+		to_chat(xeno, SPAN_WARNING("\The [src] is already broken!"))
+		return TRUE
```
Comment (stanalbatross) [inline code/modules/power/lighting.dm:597]: ?

**Review by stanalbatross — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/abilities/xeno_action.dm [hunk 4]
```diff
@@ -235,7 +252,7 @@
 // Checks whether the action is on cooldown. Should not be overridden.
 // Returns TRUE if the action can be used and FALSE otherwise.
 /datum/action/xeno_action/proc/action_cooldown_check()
-	return (cooldown_timer_id == TIMER_ID_NULL)
+	return (cooldown_timer_id == TIMER_ID_NULL) && (charge_time == null || charge_ready)
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/xenomorph/abilities/xeno_action.dm:255]: !charge_time?

**Review by stanalbatross — APPROVED** [review]

**Review by stanalbatross — APPROVED** [review]

**Pull request merged**